modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -5069,6 +5180,10 @@ private void processNodeLeftMessage(TcpDiscoveryNodeLeftMessage msg) {
             }
 
             if (msg.verified() && !locNodeId.equals(leavingNodeId)) {
+                //we will need to recalculate this value since the topology changed
+                clusterSupportsTcpDiscoveryNodeSerializationOptimization =
+                    allNodesSupport(TCP_DISCOVERY_MESSAGE_NODE_SERIALIZATION_OPTIMIZATION);
```

Comment:
No need to recalculate this property if node leaves the cluster that already supports the feature.
The same applies to NODE_FAILED event.